src/decorator-annotator.ts

```diff
@@ -15,14 +15,23 @@ import {SourceMapper} from './source_map_utils';
 import {assertTypeChecked, TypeTranslator} from './type-translator';
 import {toArray} from './util';
 
+/**
+ * ConstructorParameters are gathered from constructors, so that their type information and
+ * decorators can later be emitted as an annotation.
+ */
+interface ConstructorParameter {
+  type: ts.TypeNode|null;
```

Comment:
This could probably use a bit of docs, because it's pretty weird that it's a TypeNode (reference to ast) and also that we only set it if the AST refers to a value.